lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -716,67 +712,29 @@ def accessible_objects(user_or_attribute, operation, options = {})
 
     private
 
-    def relevant_associations(user_or_attribute, operation, object_or_attribute)
-      if operation.is_a?(class_for_type('operation'))
-        associations_between(user_or_attribute, object_or_attribute).where(id: operation.policy_element_associations)
-      else
-        associations_between(user_or_attribute, object_or_attribute).joins(:operations).where(policy_elements: {unique_identifier: operation})
-      end
+    def prohibition_for(operation)
+      operation_id = operation.try(:unique_identifier) || operation.to_s
+      PolicyMachineStorageAdapter::ActiveRecord::Operation.find_by_unique_identifier("~#{operation_id}")
     end
 
-    def is_privilege_single_policy_class(user_or_attribute, operation, object_or_attribute)
+    def accessible_operations(user_or_attribute_id, object_or_attribute_id, operation_id = nil)
       transaction_without_mergejoin do
-        relevant_associations(user_or_attribute, operation, object_or_attribute).exists?
-      end
-    end
+        user_attribute_ids = Assignment.descendants_of(user_or_attribute_id).pluck(:id) | [user_or_attribute_id]
+        object_attribute_ids = Assignment.descendants_of(object_or_attribute_id).pluck(:id) | [object_or_attribute_id]
 
-    def is_privilege_multiple_policy_classes(user_or_attribute, operation, object_or_attribute, policy_classes_containing_object)
-      transaction_without_mergejoin do
-        #Outstanding active record sql adapter prevents chaining an additional where using the association.
-        # TODO: fix when active record is fixed
-        policy_classes_containing_object.all? do |pc|
-          if operation.is_a?(class_for_type('operation'))
-            associations_between(user_or_attribute, object_or_attribute).where(id: operation.policy_element_associations.to_a, object_attribute_id: pc.ancestors).any?
-          else
-            associations_between(user_or_attribute, object_or_attribute).joins(:operations).where(policy_elements: {unique_identifier: operation}, object_attribute_id: pc.ancestors).any?
-          end
-        end
-      end
-    end
+        associations =
+          PolicyElementAssociation.where(
```

Comment:
could we filter this down farther by filtering this down with relevant opset ids when provided?  I guess at that point all we'd want is a count greater than one, no?  seems like that would be a nice perf boost with a limit clause for very authorized users